apps/src/templates/teacherDashboard/teacherSectionsRedux.js

```diff
@@ -60,17 +76,68 @@ export default function teacherSections(state=initialState, action) {
   }
 
   if (action.type === SET_SECTIONS) {
+    const assignmentList = assignments(state);
+    const sections = action.sections.map(section =>
+      sectionFromServerSection(section, assignmentList, state.studioUrl));
     return {
       ...state,
-      sectionIds: action.sections.map(section => section.id),
-      sections: _.keyBy(action.sections, 'id')
+      sectionIds: sections.map(section => section.id),
+      sections: _.keyBy(sections, 'id')
+    };
+  }
+
+  // TODO: write tests for this
```

Comment:
This is actually a TODO that should be addressed before I merge (though feel free to review still)